.travis.yml

```diff
@@ -14,62 +14,146 @@ python:
-  # `pip install -r requirements.txt` does not work on jython 2.7.1b2
+  - |
+        # install PyYAML on Jython manually before pip tries to do it and fails
+        if [ -n "$JYTHON" -a ! -f "$HOME/.virtualenv/jython/Lib/site-packages/PyYAML-3.11-py2.7.egg-info" ]; then
+            wget https://pypi.python.org/packages/source/P/PyYAML/PyYAML-3.11.tar.gz
+            tar -zxf PyYAML-3.11.tar.gz
+            cd PyYAML-3.11/
+            wget http://pyyaml.org/raw-attachment/ticket/163/jython-setup.patch
+            patch < jython-setup.patch
+            python setup.py install
+            cd -
+        fi
+
   # installing regex on jython fails (jython can't compile the native extension), this awk filter prevents it
   - cat requirements*.txt | awk \"${JYTHON:-False}'" == "True" && /^regex/ || ! /^regex/' | xargs pip install
+
+  # custom things specific to whylog-vim
   - if [ -z "$SCA" ]; then python setup.py develop; else echo skip; fi
   - git clone https://github.com/9livesdata/whylog.git; cd whylog; python setup.py develop; cd -
 
 before_script:
+  - pip freeze
   # Before runing the test case, we need to make jython run some code as in first run it can put something on stdout
   - if [ -n "$JYTHON" ]; then python -c "print ''"; else echo skip; fi
 
 script:
-  - if [ -n "$SCA" -a -n "$TRAVIS_COMMIT_RANGE" ]; then git diff --name-only "$TRAVIS_COMMIT_RANGE" | grep '\.py$' | xargs yapf --diff setup.py || exit; else echo skip; fi
-  - if [ -n "$SCA" -a -z "$TRAVIS_COMMIT_RANGE" ]; then yapf --diff --recursive . || exit; else echo skip; fi
+  # nosetests
+  - if [ -z "$SCA" ]; then python setup.py nosetests; else echo skip; fi
+
+  # SCA
+  - |
+        if [ -n "$SCA" ]; then
+            if [ -n "$TRAVIS_COMMIT_RANGE" ]; then
+                echo "using commit range: $TRAVIS_COMMIT_RANGE"
+                git diff --name-only "$TRAVIS_COMMIT_RANGE" | grep '\.py$' | xargs yapf --diff setup.py || exit
+            else
+                yapf --diff --recursive . || exit
+            fi
+        else
+            echo skip
+        fi
   - if [ -n "$SCA" ]; then pyflakes . || exit; else echo skip; fi
-  - if [ -n "$SCA" ]; then grep -r '\_\_metaclass\_\_' whylog_vim/ && exit 1 || true; else echo skip; fi
-  - if [ -n "$SCA" ]; then exit; else echo skip; fi
-  - python setup.py nosetests
+
+  # isort doesn't get along with yapf. Hopefully we'll figure it out and re-enable in the future.
+  # - if [ -n "$SCA" ]; then isort --order-by-type --recursive --line-width 100 --diff --verbose --check-only || exit; else echo skip; fi
+
+  # home-made bad code detectors
+  - if [ -n "$SCA" ]; then grep -r --extended-regex '^ *class [^\(]+(\(\))?:' whylog/ && exit 1 || true; else echo skip; fi  # old-style class detector
+  - if [ -n "$SCA" ]; then grep -r 'isinstance\(' . && exit 1 || true; else echo skip; fi  # very non-pythonic
```

Comment:
Why shouldn't it take `.`? This way it is generic and usable in the same form in all repositories.